### DISCUSSION TITLE
The background color is not changed when toggling from darkAlgorithm to defaultAlgorithm, it's a bug? how should it be done?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [ant-design/ant-design#44843] [Q&A]
CodeSandbox : https://codesandbox.io/s/change-theme-dh5x6n?file=/app.tsx:389-405
<details>
<summary>Example</summary>

```tsx
import { useState } from "react";
import { ConfigProvider, theme, Button, Card } from "antd";

function App() {
 const { defaultAlgorithm, darkAlgorithm } = theme;
 const [isDarkMode, setIsDarkMode] = useState(false);

 const handleClick = () => {
  setIsDarkMode((previousValue) => !previousValue);
 };

 return (
  <ConfigProvider
   theme={{
    algorithm: isDarkMode ? darkAlgorithm : defaultAlgorithm,
   }}>
   <Card style={{ width: "max-content" }}>
    <Button onClick={handleClick}>
     Change Theme to {isDarkMode ? "Light" : "Dark"}
    </Button>
   </Card>
  </ConfigProvider>
 );
}

export default App;

```
</details>

As you can see the background color does not change, what should I do to make it change?

___

For example, in the example of the docs where they use the `darkAlgorithm` I see that the background color is put in the CSS in the docs example.

```css
html, body {
  background: #000;
}
````

Shouldn't it automatically be black/white using the corresponding `algorithm` in the `ConfigProvider`?

Example from docs:
https://codesandbox.io/s/hp9p22?file=/index.tsx

docs:
https://ant.design/docs/react/customize-theme#use-preset-algorithms

___

Checking the antd-style examples has the same "problem" with the background

https://ant-design.github.io/antd-style/guide/switch-theme

https://codesandbox.io/s/myd3fd

___

Thanks

**Comment by MadCcc** [comment]
antd will not modify styles of `html` or `body` because they do not have a scope, all themes will affect them if we did this.

So you should modify background of body in your app.

**Reply by MiniSuperDev** [reply]
Thanks, for example if I use the `ThemeProvider` of antd-style and I want to use the `appearance` or `isDarkMode` value of the theme, how do you recommend me to update that background color for the corresponding color in light and dark theme?

In other UI frameworks this is automatically when the theme changes, I don't know if they change the background color of wrap all the app in a container, but works out of the box.

Thanks 

https://ant-design.github.io/antd-style/guide/switch-theme
